### ISSUE TITLE
Vintage Engine mistakenly executes test methods for non-included/excluded tags

### ISSUE THREAD

**Issue by marcphilipp** [junit-team/junit5#1265]
When using a TagFilter to include/exclude a tag that represents a JUnit 4 category, e.g. `com.acme.Foo`, the Vintage Engine correctly returns a filtered tree of `TestDescriptors` after discovery. However, when executing these `TestDescriptors`, Vintage will run all test methods in a class that contains an included test method.

For example, given the following test class

```java
public class ExampleTests {
    @Test
    @Category(Foo.class)
    public void foo() {}

    @Test
    public void bar() {}
}    
```

and an `includeTags("com.acme.Foo")` filter, Vintage executes both `foo` and `bar`. Similarly, for an `excludeTags("com.acme.Foo")` filter, Vintage runs both methods.

## Deliverables

- [ ] Ensure `PostDiscoveryFilters` are taken into account when executing a test class

**Comment by blindpirate** [comment]
I've encountered this issue with `5.0.3` just now. It's resolved by `5.1M2`. Could you let me know when `5.1M2` will release? Thanks.

**Comment by marcphilipp** [comment]
See https://github.com/junit-team/junit5/issues/1276#issuecomment-362499592.
